profiler/src/ProfilerEngine/Datadog.Profiler.Native/CorProfilerCallback.cpp

```diff
@@ -576,8 +618,36 @@ HRESULT STDMETHODCALLTYPE CorProfilerCallback::Initialize(IUnknown* corProfilerI
         return E_FAIL;
     }
 
+    _pConfiguration = std::make_unique<Configuration>();
+
     // Log some more important environment info:
-    CorProfilerCallback::InspectRuntimeVersion(_pCorProfilerInfo);
+    USHORT major = 0;
+    USHORT minor = 0;
+    CorProfilerCallback::InspectRuntimeVersion(_pCorProfilerInfo, major, minor);
+
+    // CLR events-based profilers need ICorProfilerInfo12 (i.e. .NET 5+) to setup the communication.
+    // If no such provider is enabled, no need to trigger it.
+    // TODO: update the test when a new CLR events-based profiler is added (contention, GC, ...)
+    if ((major >= 5) && _pConfiguration->IsAllocationProfilingEnabled())
```

Comment:
I think we should log a warning if `IsAllocationProfilingEnabled` is true but the runtime is not .NET 5+

Reply:
✅ done